osutil/disks/labels.go

```diff
@@ -39,10 +40,98 @@ func BlkIDEncodeLabel(in string) string {
 		case utf8.RuneLen(r) > 1:
 			buf.WriteRune(r)
 		case !strings.ContainsRune(allowed, r):
-			fmt.Fprintf(buf, `\x%x`, r)
+			fmt.Fprintf(buf, `\x%02x`, r)
 		default:
 			buf.WriteRune(r)
 		}
 	}
 	return buf.String()
 }
+
+type blkIdDecodeState int
+
+const (
+	stNormal blkIdDecodeState = iota
+	stSlashEscape
+	stSlashEscapeX
+	stSlashEscapeXNum
+)
+
+// BlkIDDecodeLabel decodes a string such as a filesystem or partition label
+// encoded by udev in BlkIDEncodeLabel for normal comparison, i.e.
+// "BIOS\x20Boot" becomes "BIOS Boot"
+func BlkIDDecodeLabel(in string) (string, error) {
+	out := strings.Builder{}
+	escapedHexDigits := []rune{}
+	st := stNormal
+	for _, r := range in {
```

Comment:
While I cannot point out at any deficiencies in this function, it seems to me that we could do the same in way fewer lines by using `regexp.ReplaceAllStringFunc()` to replace all the hexadecimal sequencies, and then return the parsing error if the resulting string still contains a `\`.